src/lumen/activity/field.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include "lumen/activity/activity.hpp"
+
+#include <string>
+
+namespace lumen::activity {
+
+class Activity;
+
+/// Base class from which displayable types derive.
+template <typename T>
+class Field {
+public:
+    /** Field constructor.
+     *
+     * \param parent Pointer to the activity object that should be updated when
+     * a state change occurs.
+     *
+     * \param value Initial value of the field.
+     */
+    Field(Activity* parent, T value) : parent_{parent}, value_{value} {}
+    virtual ~Field() = default;
+
+    /// Get the current value of the field.
+    [[nodiscard]] T get_value() const
+    {
+        return value_;
+    }
+
+    /// Set the value of the field and trigger a display update.
+    void set_value(T value)
+    {
+        value_ = value;
+        signal_update();
+    }
+
+    /// Convert the contents of the field to a string.
+    virtual std::string to_string() = 0;
```

Comment:
```suggestion
    virtual std::string to_string() const = 0;
```

Reply:
Not entirely sure if this would work, but it would make sense to do so since we are not changing the object.